collect_app/src/main/java/org/odk/collect/android/entities/JsonFileEntitiesRepository.kt

```diff
@@ -15,8 +15,18 @@ class JsonFileEntitiesRepository(directory: File) : EntitiesRepository {
         return readJson().keys
     }
 
-    override fun getEntities(list: String): List<Entity> {
-        return readEntities().filter { it.list == list }
+    override fun getEntities(list: String): List<Entity.Saved> {
+        return readEntities().filter { it.list == list }.mapIndexed { index, entity ->
```

Comment:
This sort of thing is obviously going to be much more awkward to implement in SQL, but I think we have some options. Worst case, we'll need to do more work in form download (or in some new background job kicked off after form download) to give ourselves an "index".